### PR TITLE
[COMDLG32][WINSPOOL] Fix crashes in winetests for comdlg32:printdlg

### PR DESCRIPTION
**Pull request by Doug-Lyons**
## Purpose

_Fix crashes in winetest comdlg32:printdlg._
JIRA issue: [CORE-19173](https://jira.reactos.org/browse/CORE-19173)

## Proposed changes

_Use PSEH to check for bad memory parameters and test for NULL parameters in print functions._

Testbot results for comdlg32:printdlg:
printing_crash_fix.patch JID66371 on top of 0.4.15-dev-6611-ge4512e6

VBox: https://reactos.org/testman/compare.php?ids=89526,89546 LGTM
KVM: https://reactos.org/testman/compare.php?ids=89525,89547 LGTM

### CODE REVIEW

File: dll/win32/comdlg32/printdlg.c [hunk 1]
```diff
@@ -3093,6 +3127,12 @@ static BOOL pagesetup_change_printer(LPWSTR name, pagesetup_data *data)
     DocumentPropertiesW(0, 0, name, dm, NULL, DM_OUT_BUFFER);
 
     pagesetup_set_devmode(data, dm);
+#ifdef __REACTOS__
+    if(!data->u.dlgw->hDevMode)
+    {    ERR("pagesetup_set_devmode failed\n");
+         goto end;
```
Comment (binarymaster) [inline dll/win32/comdlg32/printdlg.c:3133]:
```suggestion
    if (!data->u.dlgw->hDevMode)
    {
        ERR("pagesetup_set_devmode failed\n");
        goto end;
```

File: win32ss/printing/base/winspool/printers.c [hunk 2]
```diff
@@ -599,6 +599,11 @@ DeviceCapabilitiesW(LPCWSTR pDevice, LPCWSTR pPort, WORD fwCapability, LPWSTR pO
             {
                 iDevCap = fpDeviceCapabilities( hPrinter, (PWSTR)pDevice, fwCapability, pOutput, (PDEVMODE)pDevMode );
             }
+            else
+            {
+                FIXME("Handle ReactOS Not Able to Print Wine Tests\n");
```
Comment (binarymaster) [inline win32ss/printing/base/winspool/printers.c:604]: Please do not capitalize each word in a sentence.

File: dll/win32/comdlg32/printdlg.c (hunk 1 above)
Comment (Doug-Lyons) [reply dll/win32/comdlg32/printdlg.c]: Corrected. Thanks.

File: dll/win32/comdlg32/printdlg.c [hunk 3]
```diff
@@ -3408,7 +3449,15 @@ static BOOL pagesetup_wm_command(HWND hDlg, WPARAM wParam, LPARAM lParam, pagese
             INT length = SendDlgItemMessageW(hDlg, id, CB_GETLBTEXTLEN, index, 0);
             name = HeapAlloc(GetProcessHeap(),0,sizeof(WCHAR)*(length+1));
             SendDlgItemMessageW(hDlg, id, CB_GETLBTEXT, index, (LPARAM)name);
+#ifdef __REACTOS__
+            if (pagesetup_change_printer(name, data))
+            {
+                ERR("pagesetup_change_printer failed.\n");
```
Comment (JoachimHenze) [inline dll/win32/comdlg32/printdlg.c:3455]:
```suggestion
                ERR("pagesetup_change_printer failed\n");
```

File: dll/win32/comdlg32/printdlg.c [hunk 4]
```diff
@@ -3883,7 +3932,15 @@ static BOOL pagesetup_common(pagesetup_data *data)
             COMDLG32_SetCommDlgExtendedError(PDERR_NODEFAULTPRN);
             return FALSE;
         }
+#ifdef __REACTOS__
+        if (!pagesetup_change_printer(def, data))
+        {
+            ERR("pagesetup_change_printer failed.\n");
```
Comment (JoachimHenze) [inline dll/win32/comdlg32/printdlg.c:3938]:
```suggestion
            ERR("pagesetup_change_printer failed\n");
```

File: dll/win32/comdlg32/printdlg.c (hunk 3 above)
Comment (Doug-Lyons) [reply dll/win32/comdlg32/printdlg.c]: Changed. Thanks.

File: dll/win32/comdlg32/printdlg.c (hunk 4 above)
Comment (Doug-Lyons) [reply dll/win32/comdlg32/printdlg.c]: Changed. Thanks.

**Comment by JoachimHenze** [comment]
My review comments have been addressed properly. Still I will remain only neutral for this PR, because I have the feeling that it is not the optimal approach. I will not actively reject it neither. What i specifically dislike is the added  ${PSEH_LIB}).

**Pull request closed without merging**